src/components/FilterPresetsSelect/FilterPresetsSelect.tsx

```diff
@@ -0,0 +1,196 @@
+import { commonMessages } from "@dashboard/intl";
+import { Tooltip } from "@saleor/macaw-ui";
+import {
+  Box,
+  Button,
+  Dropdown,
+  DropdownButton,
+  List,
+  PlusIcon,
+  sprinkles,
+  Text,
+  vars,
+} from "@saleor/macaw-ui/next";
+import React, { MouseEvent } from "react";
+import { FormattedMessage, useIntl } from "react-intl";
+
+import { FilterPresetItem } from "./FilterPresetItem";
+import { messages } from "./messages";
+import { getSeparatorWidth } from "./utils";
+
+interface FilterPresetsSelectProps {
+  activePreset?: number;
+  savedPresets: string[];
+  selectAllLabel: string;
+  isOpen?: boolean;
+  presetsChanged?: boolean;
+  onSave: () => void;
+  onSelectAll: () => void;
+  onRemove: (filterIndex: number) => void;
+  onUpdate: (tabName: string) => void;
+  onSelect: (filterIndex: number) => void;
+  onOpenChange?: (isOpen: boolean) => void;
+}
+
+export const FilterPresetsSelect = ({
+  onSelect,
+  onRemove,
+  onSave,
+  savedPresets,
+  activePreset,
+  onSelectAll,
+  selectAllLabel,
+  isOpen,
+  onUpdate,
+  onOpenChange,
+  presetsChanged,
+}: FilterPresetsSelectProps) => {
+  const intl = useIntl();
+  const showUpdateButton =
+    presetsChanged && savedPresets.length > 0 && activePreset;
+  const showSaveButton = presetsChanged;
+
+  const getLabel = () => {
+    if (!activePreset) {
+      return selectAllLabel;
+    }
+
+    return savedPresets[activePreset - 1];
+  };
+
+  const handleSelectPreset = (e: MouseEvent<HTMLElement>, index: number) => {
+    const target = e.target as HTMLElement;
+    // Prevent run onSelect when click on remove button
+    if (!["LI", "SPAN"].includes(target.tagName)) {
+      return;
+    }
+
+    onSelect(index);
+  };
+
+  const renderDropdown = () => {
+    if (!savedPresets?.length) {
+      return (
+        <Box display="flex" alignItems="center">
+          <Tooltip title={intl.formatMessage(messages.noPresets)}>
+            <Text variant="title" size="small">
+              {selectAllLabel}
+            </Text>
+          </Tooltip>
+        </Box>
+      );
+    }
+
+    return (
+      <Dropdown
+        open={isOpen}
+        onOpenChange={open => {
+          if (onOpenChange) {
+            onOpenChange(open);
+          }
+        }}
+      >
+        <Dropdown.Trigger>
+          <DropdownButton
+            variant="text"
+            size="medium"
+            data-test-id="show-saved-filters-button"
+            style={{
+              borderColor: isOpen ? vars.colors.border.brandDefault : undefined,
+            }}
+          >
+            <Box __maxWidth="200px">
+              <Text ellipsis variant="title" size="small" display="block">
+                {getLabel()}
+              </Text>
+            </Box>
+          </DropdownButton>
+        </Dropdown.Trigger>
+        <Dropdown.Content align="start">
+          <List
+            __maxWidth={250}
```

Comment:
**issue**: this is tricky, any higher token values and your dropdown will be broken, try to avoid hardcoded values.